src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -125,6 +141,22 @@ public void addStudent(Student student) {
         csBook.addStudent(student);
     }
 
+    @Override
+    public Student getStudentByName(Name studentName) {
+        updateFilteredStudentList(new NameContainsKeywordsPredicate(List.of(studentName.toString())));
+
+        // return null if the group is not found
+        if (getFilteredStudentList().isEmpty()) {
+            updateFilteredStudentList(PREDICATE_SHOW_ALL_STUDENTS);
+            return null;
+        }
+
+        Student retrievedStudent = getFilteredStudentList().get(0);
```

Comment:
Might want to add an assert statement here in future to check that the `filteredStudentList` has only 1 element.